sway/input/seatop_down.c

```diff
@@ -41,7 +41,7 @@ static void handle_button(struct sway_seat *seat, uint32_t time_msec,
 static void handle_pointer_motion(struct sway_seat *seat, uint32_t time_msec) {
 	struct seatop_down_event *e = seat->seatop_data;
 	struct sway_container *con = e->con;
-	if (seat_is_input_allowed(seat, con->view->surface)) {
+	if (con && seat_is_input_allowed(seat, con->view->surface)) {
```

Comment:
Hm. So this skips sending the motion events when we don't have a container? But it seems to me like we should still send the events to the layer surface?

Reply:
True. Using ```!con ||``` would fix that but skip the check for an ```exclusive_client``` for layer surfaces. However, there is also no check in the initial button press event. Is it safe to assume a layer surface can always receive motion events if it was clicked?

Reply:
> However, there is also no check in the initial button press event.

That sounds like a bug, separate from this PR. There should be a check.

Reply:
I wonder. My gut feelings says we should keep this check, maybe by replacing `e->con` with a `struct wlr_surface *`.

Reply:
That'd be ideal, if we could get something like `ref_con_l[x|y]` out of the `wlr_surface` to keep movement working.